src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Serialization/JsonReaderExtensions.cs

```diff
@@ -55,20 +55,36 @@ public static string ReadNextStringProperty(this JsonReader reader, string prope
             throw new JsonSerializationException($"Could not find string property '{propertyName}'.");
         }
 
-        public static IReadOnlyList<T> ReadPropertyArray<T>(this JsonReader reader, JsonSerializer serializer, string expectedPropertyName)
+        public static IReadOnlyList<T> ReadPropertyArray<T>(this JsonReader reader, JsonSerializer serializer, string propertyName)
             where T : class
         {
-            // Ensure we can read the property name
-            if (!reader.ReadTokenAndAdvance(JsonToken.PropertyName, out var propertyName) ||
-                propertyName.ToString() != expectedPropertyName)
+            do
             {
-                return Array.Empty<T>();
-            }
+                switch (reader.TokenType)
+                {
+                    case JsonToken.PropertyName:
+                        Debug.Assert(reader.Value.ToString() == propertyName);
+                        if (reader.Read())
+                        {
+                            return reader.ReadArray<T>(serializer);
+                        }
+                        else
+                        {
+                            return Array.Empty<T>();
+                        }
+                }
+            } while (reader.Read());
 
+            throw new JsonSerializationException($"Could not find array property '{propertyName}'.");
+        }
+
+        public static IReadOnlyList<T> ReadArray<T>(this JsonReader reader, JsonSerializer serializer)
+            where T : class
+        {
             // Ensure we're at the start of an array
             if (!reader.ReadTokenAndAdvance(JsonToken.StartArray, out _))
             {
-                return Array.Empty<T>();
+                throw new JsonSerializationException($"Invalid array structure, missing StartArray token.");
```

Comment:
If we can get the current position off the JsonReader and include it here that would be helpful in debugging.

Reply:
Updated with `Got '{reader.TokenType}'.` @NTaylorMullen just confirming this wouldn't be at odds with any logging restrictions?

Reply:
Line/Position don't seem to be possible with the [`JsonReader`](https://www.newtonsoft.com/json/help/html/Properties_T_Newtonsoft_Json_JsonReader.htm), but we could leverage [`Path`](https://www.newtonsoft.com/json/help/html/P_Newtonsoft_Json_JsonReader_Path.htm) or [`Depth`](https://www.newtonsoft.com/json/help/html/P_Newtonsoft_Json_JsonReader_Depth.htm). Would need to verify this wouldn't be at odds with any logging restrictions.

Reply:
Ah, well if it's not allowed/not available then no biggy.

Reply:
Token type shouldn't be an issue. Discussed offline, potential telemetry is sanitized.